pymc_marketing/clv/models/beta_geo.py

```diff
@@ -277,9 +316,15 @@ def expected_num_purchases(
 
         Returns
         -------
-        xr.DataArray
+        xarray.DataArray
             Expected number of purchases in the next time interval of length :math:`t`.
         """
+        warnings.warn(
+            "Deprecated method. Use 'expected_purchases' instead.",
+            FutureWarning,
+            stacklevel=1,
+        )
+
```

Comment:
If this method is meant to be the same / replacement of `expected_num_purchases`, why not just use that method's logic now and not have it in two different places

Reply:
I think it is because of the data input format (one gets arrays, and another gets a data frame). Still, in the next iteration, we will harmonize this. See https://github.com/pymc-labs/pymc-marketing/pull/709/files/b692cf3e123acd5ee2e36ac4020430e71ce414d7#r1625726624

Reply:
Yes; the new API uses a single dataframe parameter, but the old array logic is still expected in the plotting and `customer_lifetime_value` methods. Those will be given separate , follow-up PRs to keep changes focused and easier to review.